router/src/lib.rs

```diff
@@ -956,48 +966,72 @@ pub fn default_tool_prompt() -> String {
     "\nGiven the functions available, please respond with a JSON for a function call with its proper arguments that best answers the given prompt. Respond in the format {name: function name, parameters: dictionary of argument name and its value}.Do not use variables.\n".to_string()
 }
 
-#[derive(Clone, Debug, Deserialize, PartialEq, Serialize, ToSchema)]
-#[schema(example = "auto")]
-/// Controls which (if any) tool is called by the model.
-pub enum ToolType {
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(tag = "type")]
+pub enum TypedChoice {
+    #[serde(rename = "function")]
+    Function { function: FunctionName },
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct FunctionName {
+    pub name: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema, Default)]
+#[serde(from = "ToolTypeDeserializer")]
+pub enum ChatCompletionToolChoiceOption {
```

Comment:
Can we avoid java naming please ?

Reply:
yea I can revert the name change if thats better.

The name change is to align with the naming that OpenAI is using in their spec. https://github.com/openai/openai-openapi/blob/d033c364c6574068ee89f3d5f845a4830bddd503/openapi.yaml#L9581, I aligned the names because I thought it may be helpful in automatically checking the spec, however it is a very long name..

Reply:
> I thought it may be helpful in automatically checking the spec

Then let's wait for it to be actually relevant, it's not usually a good thing to do things "because we might need it". Either you need it or you don't.

Personally I don't think it has any relevance to use their names.

Reply:
got it that make sense, I've changed the name to `ToolChoice` in the latest commit. (its much cleaner 👍)